Content.Server/_ES/Masks/Traitor/ESAddMaskOnUseSystem.cs

```diff
@@ -0,0 +1,104 @@
+using Content.Shared.Popups;
+using Microsoft.CodeAnalysis;
+using Robust.Shared.Player;
+
+namespace Content.Server._ES.Masks.Traitor;
+
+public sealed class ESAddMaskOnUseSystem : EntitySystem
+{
+    [Dependency] private readonly ESMaskSystem _mask = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doafter = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SharedMindSystem _mind = default!;
+    [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ESAddMaskOnUseComponent, AfterInteractEvent>(OnInteract);
+        SubscribeLocalEvent<ESAddMaskOnUseComponent, ESAddMaskOnUseDoAfterEvent>(OnDoAfter);
+    }
+
+    private void OnInteract(EntityUid uid, ESAddMaskOnUseComponent component, ref AfterInteractEvent args)
+    {
+        if (args.Target == null)
+            return;
+
+        if (!_mobState.IsCritical((EntityUid)args.Target) && component.RequireCrit)
+            return;
+
+        if (HasComp<MindShieldComponent>(args.Target) && component.MindshieldPrevent)
+            return;
+
+        if (!_mind.TryGetMind((EntityUid)args.Target!, out var mind, out var mindComponent)) // No SSD people
+            return;
+
+        _mask.TryGetTroupe((mind, mindComponent), out var troupe);
+
+        if (troupe == component.TroupeToAdd)
+            return;
+
+        if (component.Used)
+        {
+            _popup.PopupEntity(Loc.GetString("subverter-chip-used"), args.User, args.User, PopupType.Medium);
+            return;
+        }
+
+        var DoAfterArgs = new DoAfterArgs(EntityManager, args.User, component.Delay, new ESAddMaskOnUseDoAfterEvent(), eventTarget: uid, args.Target, used: uid)
+        {
+            BlockDuplicate = true,
+            BreakOnMove = true,
+            BreakOnDamage = true,
+            BreakOnDropItem = true,
+            NeedHand = true,
+            MovementThreshold = 0.5f
+        };
+
+        _doafter.TryStartDoAfter(DoAfterArgs);
+
+        _popup.PopupEntity(Loc.GetString("subverter-chip-implanting"), uid, PopupType.MediumCaution);
+
+    }
+    private void OnDoAfter(EntityUid uid, ESAddMaskOnUseComponent component, ESAddMaskOnUseDoAfterEvent args)
+    {
+        if (args.Cancelled || args.Handled)
+            return;
+
+        var target = (EntityUid)args.Target!;
+
+        if (_mobState.IsCritical(target) && component.RequireCrit)
+        {
+            _damageableSystem.SetAllDamage(target, 0);
+        }
+
+        if (!_mind.TryGetMind(target, out var mind, out var mindComponent))
+            return;
+
+        if (!_mask.TryGetTroupeEntity(component.TroupeToAdd, out var troupe))
+            return;
+
+        _mask.ApplyMask((mind, mindComponent), component.MaskToAdd, (Entity<ESTroupeRuleComponent>)troupe);
```

Comment:
This doesn't actually remove the previous mask, which is a problem. Recruits shouldn't be getting objectives from other troupes, which will be a problem once the crew troupe has proper objectives. I also think it may be wise to remove their current non-traitor objectives, but I could go either way on that position.